### PR TITLE
Cherrypick into px-rel-23.10,3

### PR DESCRIPTION
**Pull request by svijaykumar-px**
**What this PR does / why we need it**: Cherrypicks the PRS - [#1334](https://github.com/libopenstorage/operator/pull/1334), [#1366](https://github.com/libopenstorage/operator/pull/1366), [#1382](https://github.com/libopenstorage/operator/pull/1382) and [#1374 ](https://github.com/libopenstorage/operator/pull/1374) into 23.10.3 release branch

### CODE REVIEW

**Comment by nikita-bhatia** [comment]
Let's make sure all the changes from 4 PRs are present in this PR, as one of the tests is failing.

File: pkg/controller/storagecluster/update.go [hunk 1]
```diff
@@ -77,13 +79,31 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string)
 		oldPodsToDelete = append(oldPodsToDelete, pod.Name)
 	}
 
+	// If px version is greater than 2.15 then delete px-api pods along with px pods
```
Comment (piyush-nimbalkar) [inline pkg/controller/storagecluster/update.go:82]: I don't think we need changes from this file anymore. We had removed it in https://github.com/libopenstorage/operator/pull/1366.

File: drivers/storage/portworx/components_test.go [hunk 2]
```diff
@@ -517,6 +542,34 @@ func TestPortworxAPIDaemonSetAlwaysDeploys(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "registry.k8s.io/pause:3.1", ds.Spec.Template.Spec.Containers[0].Image)
 
+	cluster.Status.DesiredImages.Pause = "private.repo.org/foo/bar:1.2.3"
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+	err = testutil.Get(k8sClient, ds, component.PxAPIDaemonSetName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, "private.repo.org/foo/bar:1.2.3", ds.Spec.Template.Spec.Containers[0].Image)
+
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/components_test.go:551]: This is not expected here in this PR. I guess it came from the merge conflict.

**Review by piyush-nimbalkar — APPROVED** [review]
Just remove the two unnecessary additions, then we are good.

**Comment by piyush-nimbalkar** [comment]
Changes are already merged as part of https://github.com/libopenstorage/operator/pull/1403. Closing this.

**Pull request closed without merging**